CMakeLists.txt

```diff
@@ -415,6 +415,9 @@ endif()
 
 # optionally enable cmake testing (supported only on posix)
 option(CMAKE_TESTING "Configure test targets" OFF)
+if (${PX4_CONFIG} STREQUAL "px4_sitl_test")
+	set(CMAKE_TESTING ON)
```

Comment:
I'm confused, what happened to `BUILD_TESTING`? Couldn't you use that?

Reply:
As far as I know `BUILD_TESTING` cannot be set from the outside e.g. by the wrapping makefile. I think I even looked it up and the solution was to add a cmake option (which I called `CMAKE_TESTING`) and then have `BUILD_TESTING` enabled by including CTest 3 lines below. The reason why I set any of these options "from the outside" and not based on e.g. the target is the order the scripts are ran. So either we leave it like this and enable `CMAKE_TESTING` for every target that needs it through the wrapper or we find a nicer solution.